src/lib/storage/table.cpp

```diff
@@ -16,54 +16,66 @@
 #include "utils/assert.hpp"
 
 namespace opossum {
+Table::Table(uint32_t chunk_size) : _max_chunk_size(chunk_size) { _chunks.push_back(std::make_shared<Chunk>()); }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(row_count() == 0, "The table already contains data so adding a column is not possible.");
+
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+
+  auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(type);
+
+  _chunks.front()->add_segment(segment);
 }
 
 void Table::append(std::vector<AllTypeVariant> values) {
-  // Implementation goes here
-}
+  DebugAssert(values.size() == column_count(), "The number of values doesn't match number of columns.");
 
-uint16_t Table::column_count() const {
-  // Implementation goes here
-  return 0;
+  if (_chunks.back()->size() >= _max_chunk_size) {
+    _chunks.push_back(std::make_shared<Chunk>());
+
+    for (const auto& _column_type : _column_types) {
+      auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(_column_type);
+
+      _chunks.back()->add_segment(segment);
+    }
+  }
+
+  _chunks.back()->append(values);
 }
 
+uint16_t Table::column_count() const { return _chunks.front()->column_count(); }
+
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  return 0;
+  uint64_t number_of_rows = 0;
+  for (const auto& chunk : _chunks) {
+    number_of_rows += chunk->size();
+  }
+  return number_of_rows;
 }
 
-ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  return ChunkID{0};
-}
+ChunkID Table::chunk_count() const { return (ChunkID)_chunks.size(); }
```

Comment:
Maybe this is an overkill, but I used a C++11-style static_cast instead of (ChunkID), which is more C-style I think. Depending on the compiler warning settings, using static_cast reduces compile output noise.

Reply:
I changed it to this version, as the linter reported such errors in other places.
Interestingly enough not here